### PR TITLE
Change cljr-add-project-dependency description

### PR DESCRIPTION
**Pull request by velios**
Fix for https://github.com/clojure-emacs/clj-refactor.el/issues/547

### CODE REVIEW

**Comment by velios** [comment]
I don’t fully understand what happened, but I probably can’t offer a fix because I’m not authorized somewhere

**Comment by vemv** [comment]
Hi! Thanks for the PR.

> I don’t fully understand what happened, but I probably can’t offer a fix because I’m not authorized somewhere

What do you mean?

**Comment by velios** [comment]
> What do you mean?
image

This is the first time I've seen such a warning, but I sure you know what to do to merge it if you find PR useful

**Comment by vemv** [comment]
Thanks!

That warn is natural since you don't have write access to this repo.

Do you have any pending commit to push?

**Comment by vemv** [comment]
...note that the original issue suggests:

> Also could maybe improve the error message a little in the case where it doesn't find the project file

which seems reasonable to me. Let me know if you'd like to give if a shot - I can assist if appreciated.

**Comment by velios** [comment]
> which seems reasonable to me. Let me know if you'd like to give if a shot - I can assist if appreciated.

Yes, thank you for the opportunity. I tried to solve the problem and updated this PR.

While solving this problem, I saw the following piece of code. Now I'm not sure it's correct to specify any specific dependency management systems. If I understand correctly, five different systems are supported. I don’t know whether it’s correct to list all five or use the impersonal word "project".

For now I've settled on
`Add a dependency project file.`
But perhaps I should have left more detailed information for the user. I will wait for your opinion to resolve this.
`Add a dependency project file(supported deps.edn, project.clj, build.boot, shadow-cljs.edn and pom.xml).
`
```elisp
(defun cljr--project-file ()
  (let ((project-dir (cljr--project-dir)))
    (or (let ((file (expand-file-name "project.clj" project-dir)))
          (and (file-exists-p file) file))
        (let ((file (expand-file-name "build.boot" project-dir)))
          (and (file-exists-p file) file))
        (let ((file (expand-file-name "deps.edn" project-dir)))
          (and (file-exists-p file) file))
        (let ((file (expand-file-name "shadow-cljs.edn" project-dir)))
          (and (file-exists-p file) file))
        (let ((file (expand-file-name "pom.xml" project-dir)))
          (and (file-exists-p file) file)))))
```

**Comment by velios** [comment]
@vemv I'm sorry for the bothering you. Could you look at the resulting solution?

File: clj-refactor.el [hunk 1]
```diff
@@ -2484,11 +2484,12 @@ possible choices. If the choice is trivial, return it."
...
             (back-to-indentation)
           (paredit-backward-down))
-        (cljr-hotload-dependency)))))
+        (cljr-hotload-dependency)))
+    (error "No project file found")))
 
 ;;;###autoload
 (defun cljr-add-project-dependency (force)
-  "Add a dependency to the project.clj file.
+  "Add a dependency to the project dependencies file.
```
Comment (vemv) [inline clj-refactor.el:2492]:
```suggestion
  "Add a dependency to the project.clj or deps.edn file.
```

(It only works with those two)

File: clj-refactor.el [hunk 2]
```diff
@@ -2472,8 +2472,8 @@ possible choices. If the choice is trivial, return it."
 
 
 (defun cljr--add-project-dependency (artifact version)
-  (let* ((project-file (cljr--project-file))
-         (deps (cljr--project-with-deps-p project-file)))
+  (if-let ((project-file (cljr--project-file))
+           (deps (cljr--project-with-deps-p project-file)))
```
Comment (vemv) [inline clj-refactor.el:2476]: `deps` should go over a vanilla `let` - else the code for project.clj fails

File: clj-refactor.el [hunk 3]
```diff
@@ -2472,8 +2472,8 @@ possible choices. If the choice is trivial, return it."
 
 
 (defun cljr--add-project-dependency (artifact version)
-  (let* ((project-file (cljr--project-file))
-         (deps (cljr--project-with-deps-p project-file)))
+  (if-let ((project-file (cljr--project-file))
```
Comment (vemv) [inline clj-refactor.el:2475]: Please also check that it ends on project.clj or deps.edn

(the other results of `(cljr--project-file)` are unsupported)

File: clj-refactor.el (hunk 2 above)
Comment (vemv) [inline clj-refactor.el:2476]: Please add an entry to the Changelog.

**Review by vemv — COMMENTED** [review]
Sure, thanks for the ping!

**Comment by velios** [comment]
I'm sorry. I will not complete this PR. I don't feel confident enough when programming in elisp. This was intended as a small change to the text variable. If anyone wants to bring this to a finished state, feel free to use my PR as a basis.

**Pull request closed without merging**